flexible_event_config.md

```diff
@@ -0,0 +1,303 @@
+
+## Default configurations
+
+Here are the default configurations for event and navigation sources. Especially for navigation sources, this illustrates why the noise levels are so high relative to event sources to maintain the same epsilon values: navigation sources have a much larger output space.
+
+
+### Default event sources
+
+```jsonc
+{
+  "trigger_specs": [
+  {
+    "trigger_data": [0, 1],
+  }],
+  "max_bucket_increments": 1
+}
+```
+
+### Default navigation sources
+
+```jsonc
+{
+  "trigger_specs": [
+  {
+    "trigger_data": [0, 1, 2, 3, 4, 5, 6, 7],
+    "event_report_windows": {
+      "end_times": [<2 days>, <7 days>, <30 days>]
+    }
+  }],
+  "max_bucket_increments": 3
+}
+```
+
+## Custom configurations: examples
+
+Below are some additional configurations outside the defaults. In all of the below examples, the user(developer) is either trading-off
+
+* reducing some dimension of the default configuration(#triggers, trigger data cardinality, #windows) for increasing another one to preserve the noise level
+* reducing some dimension of the default configuration(#triggers, trigger data cardinality, #windows) for reduced noise level
+
+
+### Reporting trigger value buckets
+
+This example configuration supports a developer who wants to optimize for value data for only one reporting window (e.g. 7 days), trading fewer reporting windows for less noise. In this example any trigger that sets `trigger_data` to a value other than 0 is ineligible for attribution.
+
+```jsonc
+{
+  "trigger_specs": [
+  {
+    "trigger_data": [0],
+    "event_report_windows": {
+      "end_times": [<7 days>]
+    },
+    "summary_window_operator": "value_sum",
+    "summary_buckets": [5, 10, 100]
+  }],
+}
+```
+
+Triggers could be registered with the value field set, which are summed up and bucketed. For example if there are three triggers within 7 days of source registrations with values 1, 3 and 4. 
+
+
+```jsonc
+{ "event_trigger_data": [{"trigger_data": "0", "value": 1}] }
+{ "event_trigger_data": [{"trigger_data": "0", "value": 3}] }
+{ "event_trigger_data": [{"trigger_data": "0", "value": 4}] }
+```
+
+The values are summed(to 8) and reported as bucket [5,9]
+
+```jsonc
+{
+  ... 
+  "trigger_summary_bucket": [5, 9]
+}
+```
+
+### Reporting trigger counts
+
+This example shows how a developer can configure a source to get a count of triggers up to 10.
+
+```jsonc
+{
+  "trigger_specs": [
+  {
+    "trigger_data": [0],
+    "event_report_windows": {
+      "end_times": [<7 days>]
+    },
+    "summary_window_operator": "count",
+    "summary_buckets": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+  }],
+}
+```
+
+Attributed triggers with `trigger_data` set to 0 get counted capped to 10. The trigger value is ignored since `summary_window_operator` is set to “count”. Supposing 4 triggers are registered and attributed to the source, the report would look like this:
```

Comment:
Clarify:  `trigger_data` set to 0 get **counted capped** to 10

Reply:
Oh actually I just realized I missed this comment! I'm a bit confused though since it seems like what you wrote is exactly the same as what's in the doc. Am I missing something?

Reply:
I didn't quite understand it so left a comment to may be clarify the sentence.  Maybe something like - Attributed triggers with `trigger_data` set to 0 are counted and capped at 10 ?

Reply:
Ah got it. Will make that change!